ServiceBus/ServiceBus/BasketService.svc.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.ServiceModel;
+using System.Text;
+using SharedLibs.DataContracts;
+
+namespace ServiceBus
+{
+    
+    public class BasketService : IBasketService
+    {
+        /// <summary>
+        /// Method is supposed to create new basket. This is a test for database FK implementation.
+        /// </summary>
+        /// <returns>DTO object basket WITHOUT basket items.</returns>
+        public SharedLibs.DataContracts.Basket CreateBasekt()
+        {
+            try
+            {
+                using (var context = new ServiceBusDatabaseEntities())
+                {
+                    //Add new basket and two campaigns
+                    //It shoudl not be possible to store into a db since
+                    //campaings guid does not exists yet and there is a FK
+
+                    Basket basket = new Basket()
+                    {
+                        Id = Guid.NewGuid()
+                    };
+
+                    var camp2 = context.Campaign.FirstOrDefault(p => p.Id == new Guid("509a8d88-8fa1-491a-94ba-fb8ed6aaf6fb"));
+                    
+                    //I kinda a thought when there are no records in campaign table it will throw error since
+                    //FK is pointing into campaign table Id. Insted new campaign was created. That is sick
+                    basket.Campaigns.Add(new Campaign() {Discount = 15, Name = "XYth campaign" , Id = Guid.NewGuid()});
+                    //var camp4 = new Campaign() { Discount = 150, Name = "4st campaign", Id = Guid.NewGuid() };
+                    //basket.Campaigns.Add(camp4);
+
+                    //this is what I expect to do to create new campaign... 
+                    //context.Campaign.Add( new Campaign() { Discount = 55, Name = "5st campaign", Id = Guid.NewGuid() });
+                    
+                    basket.Campaigns.Add(camp2);
+                    context.Basket.Add(basket);
+                    context.SaveChanges();
+
+                    //basket is saved therefore I would like to return SharedLibs.DataContracts.Basket(); Is this the way how to create it
+
+
+                    //TODO: Is there some C# function which can do this for me so I can use BasketCampaings = basket.Campaigns; like I tried below? Or I have to convert this always explicitly in some block of code?
```

Comment:
It is quite normal that DB Context is not the same as data contracts provided to clients. Mapping should be done manually, and that is entirely fine...